src/mds/CDir.cc

```diff
@@ -1773,8 +1773,14 @@ void CDir::_omap_fetched(bufferlist& hdrbl, map<string, bufferlist>& omap,
 	}
       }
     } else {
-      dout(1) << "corrupt directory, i got tag char '" << type << "' pos " << pos << dendl;
-      assert(0);
+      dout(1) << "corrupt directory, i got tag char '" << type << "' pos "
+        << pos << dendl;
+      cache->mds->clog->error() << "Corrupt directory entry '" << p->first
+        << "' in dirfrag " << *this;
+      // TODO: add a mechanism for selectively marking a path
+      // damaged, rather than marking the whole rank damaged.
+      cache->mds->damaged();
+      assert(0);  // Unreachable: damaged() respawns us
```

Comment:
I'm not sure I'm 100% onboard here. True, no other rank is going to try and open this because we are auth for it, but it's not really the MDS' state that's damaged either. I'm worried this conflation of damaged states will bite us.

Do you have any ideas or plans for marking paths rather than daemons as damaged?